Added propper support for Lets split vitamins

Duckle29 opened the pull request:
I fixed my issues with identifying master and slave half, and updated the readme to reflect the use of msys2

I also added avrdude to the path of msys2, and added it to the msys installer.

jackhumbert commented:
Nice! Rather than copying the keymaps into this, maybe they should be moved to the layouts section?

Duckle29 commented:
Do you mean something like using default_ortho_4x12?
I just copied what the original lets_split did :)

drashna commented on util/msys2_install.sh:
```diff
@@ -63,11 +67,37 @@ else
     done
 fi
 
+pacman -q -Q mingw-w64-x86_64-avrdude
```
Avrdude should be installed on the newer versions: 
https://github.com/qmk/qmk_firmware/blob/master/util/msys2_install.sh#L19

drashna commented:
Two questions: 
What's the difference between this and then let's split? 
And is there a reason that this couldn't be done as a revision of the lets split, rather than as a new board?

Duckle29 commented:
@drashna 

The main difference between this, where the firmware is concerned, is that it's a fair bit different hardware. You can see an image of the two halves here: https://imgur.com/a/m9q17

The MCU is on board, and due to a mistake on my end, there's no diode seperating VBUS from VCC, this is why I need to use a different scheme to detect which half is master or slave. I'm instead detecting if the half has been enumerated, but this isn't instant, which the vbus method is, so first each half idles as slaves. then once one side get's usb, it'll switch to master, and tell the other half to remain slave.

This means I've altered matrix.c and the split_util files quite a bit. 

I decided to make it a new keyboard because of this, as I didn't feel comfortable intruding this much on the original let's split work.

drashna commented:
Thanks.  That's the type of info I was after. 

Just a couple of issues still:
- Rename `lets_split.c` and `lets_split.h`, so they match the folder name.  As per: https://docs.qmk.fm/hardware_keyboard_guidelines.html
- Remove all of extra keymaps that aren't yours and the default. 

Also, it may be worth using a different name here, as it's a distinctly different board.  Like the Levinson, which is a backlight capable version of the Lets split.   This isn't necessary, though.

Duckle29 commented:
@drashna I renamed it to Vitamins Included, and did the same for the main header and c file.

I also removed all the extra keymaps, but I maintained the default one. If I remove the default one, where is the default keymap then defined? Do you have some documentation on that?

Is it what @jackhumbert referred to?

Thanks :)

Duckle29 commented:
Oh, and I haven't pushed these changes yet. I'm wating for some feedback on the new name :)

drashna commented:
awesome! 
And the default keymap is in ... well, `default`. 

And yeah, that's basically what Jack was talking about.  We want to keep new keyboards "clean".   

Though, by the layouts, he meant `/layouts/community/ortho_4x12`.  But I think he was thinking out loud.  Most of those keymaps would need to be edited to work right.  And there is no need for you to do that.

drashna commented on keyboards/lets_split_vitamins/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,214 @@
+#include "lets_split.h"
```
Could you replace the includes with just `#include QMK_KEYBOARD_H`?

drashna commented on keyboards/lets_split_vitamins/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,214 @@
+ * | Ctrl | GUI  |  Alt |Adjust|Lower |Space |Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = KEYMAP( \
```
Could you replace all of the `KEYMAP`s with `LAYOUT_ortho_4x12`?

Duckle29 replied:
I thought I did, maybe I didn't properly push the changes from my laptop / desktop. Or maybe my head is just ahead of what I've actually done. Will do :)

Duckle29 commented:
@drashna That should do it :)

drashna commented on keyboards/vitamins_included/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,214 @@
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistent_default_layer_set(1UL<<_QWERTY);
```
(optional) This function is no longer needed.  It's been rolled into core code, and will automatically handle audio.

https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L190

https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/config.h#L10

drashna commented on keyboards/vitamins_included/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,214 @@
+    case LOWER:
+      if (record->event.pressed) {
+        layer_on(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
```
(Optional) There is a new method to do this that works with any/all layer codes/commands.

If you want, set the RAISE/LOWER keycodes to MO(x) instead:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L41-L42
And add this block:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L181-L183

The caveat is that the Adjust layer is no longer reachable on it's own, because of how this works.   This method has no such issue, but isn't as flexible otherwise.

drashna commented on keyboards/vitamins_included/readme.md:
````diff
@@ -0,0 +1,142 @@
+Clone the `qmk_firmware` repo and navigate to its top level directory. [Once your build environment is setup](https://docs.qmk.fm/getting_started_build_tools.html), you'll be able to generate the default .hex using:
+
+```bash
+make lets_split_vitamins/rev1:default
````
Docs should probably be updated now! :)

drashna commented on keyboards/vitamins_included/rev1/rev1.c:
```diff
@@ -0,0 +1,13 @@
+#include "lets_split.h"
```
This needs to be changed.  To `rev1.h` most likely.

drashna commented on keyboards/vitamins_included/vitamins_included.c:
```diff
@@ -0,0 +1,16 @@
+#include "lets_split.h"
```
This should be changed to `vitamins_included.h`

Duckle29 commented:
And that should do it :)

drashna commented on keyboards/vitamins_included/rules.mk:
```diff
@@ -0,0 +1,75 @@
+
+LAYOUTS = ortho_4x12
+
+DEFAULT_FOLDER = lets_split/rev1
```
Sorry,  I missed this.  Should be `vitamins_included/rev1` instead!

Duckle29 replied:
aah danm, missed that too. I never compile without the sub folder :)

drashna commented:
Awesome!  Thanks!
